readthedocs/core/unresolver.py

```diff
@@ -0,0 +1,41 @@
+from urllib.parse import urlparse
+
+from django.urls import resolve as url_resolve
+from django.test.client import RequestFactory
+
+from readthedocs.proxito.middleware import map_host_to_project_slug
+from readthedocs.proxito.views.utils import _get_project_data_from_request
+
+
+def unresolve(uri):
```

Comment:
nit: making this a class will help us to extend it later if needed (similar to the resolver class)